src/main/java/com/google/research/bleth/simulator/TracingSimulation.java

```diff
@@ -0,0 +1,98 @@
+package com.google.research.bleth.simulator;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A simulation in which the global resolver estimates beacons real locations based on information received from observers.
+ * Beacons transmit their unique static IDs each round.
+ * Agents move according to a movement strategy.
+ * Observers change their awakeness states according to an awakeness strategy.
+ * Location estimation occurs each round and are based on both previous estimations and newly received information.
+ */
+public class TracingSimulation extends Simulation {
+
+    private TracingSimulation(TracingSimulationBuilder builder) {
```

Comment:
Once again I'll quote Alon:
"Prefer to move private methods (and constructors) below public methods. This way a reader of the class first sees the API they can use, and only then the implementation details".

Reply:
+1.